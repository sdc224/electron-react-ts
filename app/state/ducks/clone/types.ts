import { ProjectSchema } from 'gitlab';

export const CloneActionTypes = {
  FETCH_ALL_PROJECTS: '@@clone/FETCH_ALL_PROJECTS',
  FETCH_ALL_PROJECTS_SUCCESS: '@@clone/FETCH_ALL_PROJECTS_SUCCESS',
  FETCH_ALL_PROJECTS_ERROR: '@@clone/FETCH_ALL_PROJECTS_ERROR'
};

export interface IProject extends Response {
  name: string;
}

export interface ICloneState {
  loading: boolean;
  projects: Array<ProjectSchema>;
  error: string;
}

export interface ICloneAwareState {
  clone: ICloneState;
}
