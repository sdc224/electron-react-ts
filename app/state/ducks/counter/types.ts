import { Dispatch as ReduxDispatch, Store as ReduxStore, Action } from 'redux';

export const CounterActionTypes = {
  INCREMENT_COUNTER: '@@counter/INCREMENT_COUNTER',
  DECREMENT_COUNTER: '@@counter/DECREMENT_COUNTER'
};

export type CounterStateType = {
  counter: number;
};

export type GetState = () => CounterStateType;

export type Dispatch = ReduxDispatch<Action<string>>;

export type Store = ReduxStore<CounterStateType, Action<string>>;
