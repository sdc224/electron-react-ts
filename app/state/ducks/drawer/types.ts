export const DrawerActionTypes = {
  TOGGLE_MOBILE_OPEN: '@@drawer/TOGGLE_MOBILE_OPEN'
};

export interface IDrawerState {
  mobileOpen: boolean;
}

export interface IDrawerAwareState {
  drawer: IDrawerState;
}
