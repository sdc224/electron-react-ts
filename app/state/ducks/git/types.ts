export const GitOperationTypes = {
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR'
};

export type GitNamespace = 'remote' | 'checkout' | 'branch';

export interface IGitOperation {
  loading: boolean;
  success: unknown;
  error: string;
}

export interface IGitOperationWithNamespaceState {
  // clone: IGitOperation;
  remote: IGitOperation;
  checkout: IGitOperation;
  branch: IGitOperation;
}

export interface IGitOperationAwareState {
  gitOperation: IGitOperationWithNamespaceState;
}
