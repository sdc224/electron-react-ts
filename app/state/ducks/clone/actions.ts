import { action } from 'typesafe-actions';
import { ProjectSchema } from 'gitlab';
import { CloneActionTypes } from './types';

export const fetchAllProjects = () =>
  action(CloneActionTypes.FETCH_ALL_PROJECTS);

export const fetchAllProjectsSuccess = (data: ProjectSchema[]) =>
  action(CloneActionTypes.FETCH_ALL_PROJECTS_SUCCESS, data);

export const fetchAllProjectsError = (message: string) =>
  action(
    CloneActionTypes.FETCH_ALL_PROJECTS_ERROR,
    undefined,
    undefined,
    message
  );

export const toggleCloneProgress = () =>
  action(CloneActionTypes.TOGGLE_CLONE_PROGRESS);
