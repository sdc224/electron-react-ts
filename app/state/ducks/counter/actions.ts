import { action } from 'typesafe-actions';
import { CounterActionTypes } from './types';

export const increment = () => action(CounterActionTypes.INCREMENT_COUNTER);

export const decrement = () => action(CounterActionTypes.DECREMENT_COUNTER);
