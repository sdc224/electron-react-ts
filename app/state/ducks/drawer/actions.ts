import { action } from 'typesafe-actions';
import { DrawerActionTypes } from './types';

// eslint-disable-next-line import/prefer-default-export
export const toggleMobileOpen = () =>
  action(DrawerActionTypes.TOGGLE_MOBILE_OPEN);
