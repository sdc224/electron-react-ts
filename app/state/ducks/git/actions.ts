import { action } from 'typesafe-actions';
import { GitNamespace, GitOperationTypes } from './types';

export const start = (namespace: GitNamespace, params: unknown) =>
  action(`@@${namespace}/${GitOperationTypes.LOADING}`, params);

export const success = (namespace: GitNamespace, payload: unknown) =>
  action(`@@${namespace}/${GitOperationTypes.SUCCESS}`, payload);

export const error = (namespace: GitNamespace, message: string) =>
  action(
    `@@${namespace}/${GitOperationTypes.ERROR}`,
    undefined,
    undefined,
    message
  );
