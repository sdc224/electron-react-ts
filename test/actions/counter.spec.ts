import { action } from 'typesafe-actions';
import * as actions from '@ducks/counter/actions';
import { CounterActionTypes } from '@ducks/counter/types';

describe('counter actions', () => {
  it('should increment should create increment action', () => {
    const expectedAction = action(CounterActionTypes.INCREMENT_COUNTER);
    expect(actions.increment()).toEqual(expectedAction);
  });

  it('should decrement should create decrement action', () => {
    const expectedAction = action(CounterActionTypes.DECREMENT_COUNTER);
    expect(actions.decrement()).toEqual(expectedAction);
  });
});
