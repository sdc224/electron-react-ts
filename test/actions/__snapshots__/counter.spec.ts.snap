// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should decrement should create decrement action 1`] = `
Object {
  "type": "@@counter/DECREMENT_COUNTER",
}
`;

exports[`actions should increment should create increment action 1`] = `
Object {
  "type": "@@counter/INCREMENT_COUNTER",
}
`;
